### PR TITLE
feat: add custom network modal form

### PR DESCRIPTION
**Pull request by salimtb**
## **Description**
 the custom add network form flow is now available on the modal

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25213?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Run `yarn && ENABLE_NETWORK_UI_REDESIGN=1 yarn start`
2. Go to Settings -> Developer Options
3. Tun on the network new toggle
4. Go to the wallet page
5. Click on the network button ( see the video below )
6. you should see the list of popular network
7.click on add network ( you should see the network form )

### **Before**

https://github.com/MetaMask/metamask-extension/assets/26223211/02ab2204-8160-4b11-9d85-20c194e201b7

### **After**

https://github.com/MetaMask/metamask-extension/assets/26223211/93c8a0f3-2939-4f0f-84de-2adc9afbec42

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by bergeron — DISMISSED** [review]
worked locally, was able to add a new network from the modal

File: ui/components/app/modals/modal.js [hunk 1]
```diff
@@ -117,7 +117,7 @@ const custodyConfirmModalStyle = {
 
 const MODALS = {
   ONBOARDING_ADD_NETWORK: {
-    contents: <AddNetworkModal />,
+    contents: <AddNetworkModal isNewNetworkFlow={false} />,
```
Comment (darkwing) [inline ui/components/app/modals/modal.js:120]: Awesome, awesome catch!  I hadn't thought of this flow!

Could we provide a better prop name than `isNewNetworkFlow`?  At first glance I don't understand what it represents.

File: ui/pages/onboarding-flow/add-network-modal/index.js [hunk 2]
```diff
@@ -14,7 +15,7 @@ import {
 
 import NetworksForm from '../../settings/networks-tab/networks-form/networks-form';
 
-export default function AddNetworkModal() {
+export default function AddNetworkModal({ isNewNetworkFlow = false }) {
```
Comment (darkwing) [inline ui/pages/onboarding-flow/add-network-modal/index.js:18]: The `defaultProps` at the bottom sets this value to `true`.  Which should it be?

**Review by bergeron — DISMISSED** [review]

**Review by darkwing — CHANGES_REQUESTED** [review]
One nit and one question!  Almost perfect!

File: ui/components/app/modals/modal.js [hunk 3]
```diff
@@ -117,7 +117,7 @@ const custodyConfirmModalStyle = {
 
 const MODALS = {
   ONBOARDING_ADD_NETWORK: {
-    contents: <AddNetworkModal />,
+    contents: <AddNetworkModal newNetworkMenuDesignActive={false} />,
```
[the remark this answers is not in the document]
Comment (salimtb) [reply ui/components/app/modals/modal.js]: we don't need to use it here , nice found i just fixed it

**Review by bergeron — DISMISSED** [review]

**Review by darkwing — DISMISSED** [review]

**Review by bergeron — DISMISSED** [review]

**Review by bergeron — DISMISSED** [review]

**Review by darkwing — DISMISSED** [review]

**Review by bergeron — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**